netutils/dhcpc/dhcpc.c

```diff
@@ -458,6 +458,7 @@ static void *dhcpc_run(void *args)
   struct dhcpc_state result;
   int ret;
 
+  memset(&result, 0, sizeof(result));
```

Comment:
why need memset here?

Reply:
If the first request fails, there may be a delay before the second request.

Reply:
If so, should we move into while block to handle the fail in no first loop?

Reply:
@normanr please take a look.

Reply:
> If so, should we move into while block to handle the fail in no first loop?

Yes, updated